client/landing/stepper/declarative-flow/internals/hooks/use-step-navigation-with-tracking/index.ts

```diff
@@ -0,0 +1,33 @@
+import { recordSubmitStep } from '../../analytics/record-submit-step';
+import type { Flow, Navigate, ProvidedDependencies, StepperStep } from '../../types';
+
+interface Params< FlowSteps extends StepperStep[] > {
+	flow: Flow;
+	currentStepRoute: string;
+	navigate: Navigate< FlowSteps >;
+	steps: StepperStep[];
+}
+
+export const useStepNavigationWithTracking = ( {
+	flow,
+	currentStepRoute,
+	navigate,
+	steps,
+}: Params< ReturnType< Flow[ 'useSteps' ] > > ) => {
+	const stepNavigation = flow.useStepNavigation(
+		currentStepRoute,
+		navigate,
+		steps.map( ( step ) => step.slug )
+	);
+
+	const shouldTrackSubmit = flow.use__Temporary__ShouldTrackEvent?.( 'submit' ) ?? false;
+
+	return {
+		...stepNavigation,
+		submit: ( providedDependencies: ProvidedDependencies = {}, ...params: string[] ) => {
+			shouldTrackSubmit &&
+				recordSubmitStep( providedDependencies, '', flow.name, currentStepRoute );
+			stepNavigation.submit?.( providedDependencies, ...params );
+		},
```

Comment:
Maybe if `shouldTrackSubmit` equals false, we just return `stepNavigation` as-is?

Reply:
Think of `shouldTrack*` being the entity here. So `shouldTrackBack`, `shouldTrackNext`, etc. can all be worked out in parallel. Does this help to clarify what's intended with this?